### PR TITLE
Robust guard file creation and deletion, and some cleanups.

### PR DESCRIPTION
**Pull request by cauthmann**
Hello,

see the individual commit messages.

Some of the commits depend on others, so I'm creating a single pull request. Let me know if you want me to split it up, or if any of the changes are not to your liking.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -10,7 +10,7 @@
     "build": "tsc",
     "prepare": "npm run build",
     "lint": "eslint .",
-    "format": "eslint --fix --ext ts && prettier --write *.json *.js *.yml src/**/*.ts .github/**/*.yml",
+    "format": "eslint --fix --ext ts && prettier --write *.json *.js *.yml src/**/*.ts tests/**/*.ts .github/**/*.yml",
     "format:check": "prettier --list-different *.json *.js *.yml src/**/*.ts .github/**/*.yml"
```
Comment (rhys-vdw) [inline package.json:14]: Should add `tests/**/*.ts` to `format:check` too.

**Review by rhys-vdw — COMMENTED** [review]
Wow great work. I do wonder if we need to be quite so anal about clearing existing files and checking them. (Generally pretty safe to assume that users are using version control and can deal with these edge cases). But given the quality and thoroughness of work I'm not going to knock it.

Thanks for your contribution. I have one little request for adjustment.

Annoyingly we don't seem to have any CI checking on PRs anymore. I thought I fixed this but it seems to still be an issue. Feel free to merge after you address my comment, just make sure you've given the tests a run locally.

Thanks @cauthmann!

**Comment by rhys-vdw** [comment]
Oh, I just noticed that you didn't accept your collaborator invite: https://github.com/rhys-vdw/ts-auto-guard/invitations

**Comment by cauthmann** [comment]
My idea was that formatting errors in tests shouldn't block releases, so I didn't add it.. but of course formatting errors should block commits. Not sure what I was thinking.

I did run the tests before each commit (if only to fix the formatting that my own formatting settings destroyed).

Seeing the queued PR #129, I also added a commit to remove lodash entirely. We use modern javascript features anyway, and those make most of lodash's functionality redundant.

I did not accept the invite because I'm not going to stick around. All of my typescript projects are monorepos, so I cannot use ts-auto-guard until my other bug is fixed, which won't happen anytime soon. ts-morph is designed to immediately normalize all paths, and it isn't possible to detect whether a file is part of `node_modules` after the original path is erased. Supporting symlinks in ts-morph looks like a major undertaking.

**Pull request merged**